src/Microsoft.Azure.WebJobs.Host/Extensions/AppSettingAttributeExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    internal static class AppSettingAttributeExtensions
+    {
+        public static AutoResolveAttribute ToAutoResolve(this AppSettingAttribute attribute)
+        {
+            return new AutoResolveAttribute
+            {
+                Default = attribute.Default,
+                AllowTokens = false
```

Comment:
Is the crux of this whole proposal a way to make the terminology of AllowTokens more understandable? If so, rather than introduce a new attribute and have 2 mechanisms around (further complicating things I think), what if we instead renamed AllowTokens to IsAppSetting? @brettsam I remember when we initially named this we had a lot of back and forth on the right name - we might have even had a name like this proposed at one time.

What I don't think we want to do is keep both ways - either we remove AllowTokens completely, or we rename this. `[AutoResolve(IsAppSetting = true)]` seems clear doesn't it?